Updating documentation for the ProBuilder tab in the Preferences window (#540)
Unity-Technologies/com.unity.probuilder: Documentation~/preferences.md

# ProBuilder Preferences

To configure ProBuilder, go to **Edit** > **Preferences** (macOS: **Unity** > **Settings**) in the main menu. In the **Prefereces** window, select **ProBuilder** from the list on the left.

## Developer Mode

> Set what an element rotates around. This setting has a different effect depending on the selection mode. [[Jon I tried very hard to figure this out and couldn't; do you know?]]

Reviewer: I don't have this option. Is this option only enabled when Developer mode is enabled?

Author: That makes sense.

| **Pivote mode** | **Vertex selection mode** | **Edge selection mode** | **Face selection mode** |
| --- | --- | --- | --- |
| **Center** | | | | 
| **Individual origins** | | | |
| **Active Element** | | | |

To add the pivot point and orientation status to the **Scene** view, enable **Show Internal Pivot and Orientation**.

## Dimensions Overlay

By default, the [Dimensions overlay](menu-dimover.md) displays the dimensions of the selected mesh elements only. To display the dimension of the entire object, regardless of the selected elements (vertices, edges, or faces), change the **Bounds Display** property from **Object** to **Element**.

## Experimental

> Enable or disable ProBuilder experimental features:

Reviewer: ```suggestion
Enable or disable these ProBuilder experimental features:
```

* **Tools** > **ProBuilder** > **Editors** > [**New Bezier Shape**](bezier.md)
* **Tools** > **ProBuilder** > **Experimental** > **Boolean (CSG)**.
* **Preferences** window > **ProBuilder** > **Store Mesh as Asset**.

> **Warning:** Experimental features can have unpredictable results.

| **Property** | **Function** |
| :--- | :--- |
| **Experimental Features Enabled** | Add the experimental features to the Editor. |

> | **Store Meshes as Asset** | Store meshes as standalone assets in the Project folder, rather than as part of the Scene asset. |

Reviewer: ```suggestion
| **Store Meshes as Asset** | Store meshes as standalone assets in the Project folder, rather than as part of the Scene asset. This property is only visible when **Experimental Features Enabled** is enabled. |
```

> **Note**: When you activate or deactivate experimental features, the Unity Editor has to recompile scripts, which causes a delay before changes apply.

## General

Set some basic options for ProBuilder.

| **Property** | **Function** |
| :--- | :--- |

> | **Show Action Notifications** | Enable this option if you want ProBuilder to notify you when performing actions [[Jon: where, how, which actions?]]. |

Reviewer: It used to just display in the Scene view as a notification, but I can't get it to work anymore. Checking if this is a bug.

Reviewer: It is not a bug. It still works, but I think I broke it on my project.

Author: Okay, I can get it to give me an error when I try to cut: not enough elements selected for a cut. 

The thing is: it cuts. So it's showing me a lie.

Author: ```suggestion
| **Show Action Notifications** | Show explanations for failed actions, like cutting a mesh. |
```

Reviewer: Does it only show explanations for failed notifications? @modrimkus-unity, is this the case?

Reviewer 2: Yes, what Jon said, it's not just for failures.

Author: Ah, interesting. I couldn't find an action it was pleased with.

Reviewer: @modrimkus-unity what are rules for showing these notifications? Like, why do some actions get it and some don't?

Reviewer: ```suggestion
| **Show Action Notifications** | Enable this option to display a notification in the Scene view after you perform an action with ProBuilder. |
```

| **Vertex Size** | Set the size to render the vertex points on ProBuilder meshes in the **Scene** view. |
| **Line Size** | Set the size to render the edges on ProBuilder meshes in the **Scene** view. **Note:** On macOS, this property is only available if you use [OpenGL](https://www.opengl.org/) instead of Metal.|
| **Wireframe Size** | Set the size to render the ProBuilder mesh wireframe in the **Scene** view. **Note:** On macOS, this property is only available if you use [OpenGL](https://www.opengl.org/) instead of Metal. |

## Mesh Editing

Customize how meshes behave when you edit them.

| **Property** | **Function**  |
| :--- | :--- |
| **Auto Resize Colliders** | As you edit mesh bounds, colliders automatically resize to match the new bounds. |
| **Allow non-manifold actions** | Allow advanced behaviours of editing actions so you can create [non-manifold geometry](gloss.html#manifold-and-non-manifold-geometry), such as [bridging closed edges](Edge_Bridge.md). Note that this can destabilize your project; use them only if you are sure of their concepts and how to apply them. |

> | **Auto Update Action Preview** | Smooth update for the preview of any final result as you edit [[Jon: I'm not sure about this and also it's badly written]]. This option can lead to a slower editing experience when editing large selections. |

Reviewer: I'm not really clear what this one does either, actually. 
```suggestion
| **Auto Update Action Preview** | Smooth update for the preview of any final result as you edit. This option can lead to a slower editing experience when editing large selections. |
```

| **Collider Type** | Set the default type of [collision primitive](https://docs.unity3d.com/Manual/CollidersOverview.html) to use for new ProBuilder objects. The default is a [**Mesh Collider**](https://docs.unity3d.com/Manual/class-MeshCollider.html), which is a custom shape. To use a basic cube, select [**Box Collider**](https://docs.unity3d.com/Manual/class-BoxCollider.html). To create objects without a default collider, select **None**. |
| **Lightmap UVs Settings** | Set defaults for the standard Lightmap UVs parameters. [For details, refer to [Generating lightmap UVs](https://docs.unity3d.com/Manual/LightingGiUvs-GeneratingLightmappingUVs.html). To return to the default settings, click the **Reset** button | 

## Snap Settings

Customize how snapping behaves with ProBuilder.

| **Property** | **Function** |
| :--- | :--- |
| **Snap As Group** | All selected items keep the same relative position to each other while snapping. This is the default. Disable this option to snap each selected item to the grid independently. |

> | **Snap Axis** | Choose how vertices snap to the grid when you move them. **Active Axis**: Vertices snap only along the currently active axis [[Jon: what does "currently active" mean?]]. This is the default. **All Axes**: Vertices snap to all axes simultaneously. |

Reviewer: The active axis is the one you're selecting right now. When you activate the Move tool, you get the 3 little arrows: a green one for the Y-axis, a red one for the X-axis, and a blue one for the Z-axis. Whichever one you select is the active axis.

Reviewer: This is mostly used for vertex snapping (which you activate by holding **V**).

Author: But you can also select the "middle" and move freely. What happens then?

Reviewer: That just moves it freely, as you say. @modrimkus, any insight into this one?

Reviewer 2: While this involves element (vert/edge/face) snapping, it's not related to the vertex-vertex snapping (i.e. holding V). This pref is strictly for how elements snap to the grid (when the editor's built-in absolute grid snapping is enabled). 

- Active axis. If element is moved along an axis (constrained to currently active axis), then snapping will only take place on that axis (places where that axis hits a grid line or point).
- All axes. Even if element is moved along an axis, snapping will ignore that and pick the nearest point on grid and snap to it. 
- If in Free move (i.e. user holds shift and uses the square manipulator), snapping falls back to "All axes" no matter which preference.

Video illustrating the Active Axis vs All Axes setting.
https://github.com/user-attachments/assets/3fcbd0bd-3577-4f3b-a9b0-cf269d990107

Author: ```suggestion
| **Snap Axis** | Choose how mesh elements snap to the grid when you move them (if Snap to Grid is active). **Active Axis**: When you move an element along an axis, the element snaps to the next grid point on that axis. This is the default. **All Axes**: An element snaps to the nearest grid point, even if it's on a different axis to the one you're moving on. |
```
